foxglove_data_platform/client.py

```diff
@@ -474,16 +506,23 @@ def create_device(
             "name": device["name"],
         }
 
-    def delete_device(self, *, device_id: str):
+    def delete_device(
+        self, *, device_id: Optional[str] = None, device_name: Optional[str] = None
+    ):
         """
         Deletes an existing device.
 
         Note: you must first delete all imports from the device; see `delete_import`.
 
         :param device_id: The id of the device.
+        :param device_name: The name of the device.
         """
+        if device_name and device_id:
+            raise RuntimeError("device_id and device_name are mutually exclusive")
+        if not (device_name or device_id):
+            raise RuntimeError("device_id or device_name must be provided")
         response = requests.delete(
-            self.__url__(f"/v1/devices/{device_id}"),
+            self.__url__(f"/v1/devices/{device_name or device_id}"),
```

Comment:
this probably works but i'm not 100% sure - could we add a test for `delete_device`?